src/main/java/io/numaproj/numaflow/sinker/SinkerTestKit.java

```diff
@@ -168,28 +174,39 @@ public void onCompleted() {
                         .putAllHeaders(
                                 datum.getHeaders() == null ? new HashMap<>() : datum.getHeaders())
                         .build();
-                requestObserver.onNext(request);
+                requestObserver.onNext(SinkOuterClass.SinkRequest
+                        .newBuilder()
+                        .setRequest(request)
+                        .build());
             }
+            // send end of transmission message
+            requestObserver.onNext(SinkOuterClass.SinkRequest.newBuilder().setStatus(
+                    SinkOuterClass.SinkRequest.Status.newBuilder().setEot(true)).build());
 
             requestObserver.onCompleted();
 
-            SinkOuterClass.SinkResponse response;
+            List<SinkOuterClass.SinkResponse> outputResponses;
             try {
-                response = future.get();
+                outputResponses = future.get();
             } catch (Exception e) {
                 throw new RuntimeException(e);
             }
 
             ResponseList.ResponseListBuilder responseListBuilder = ResponseList.newBuilder();
-            for (SinkOuterClass.SinkResponse.Result result : response.getResultsList()) {
-                if (result.getStatus() == SinkOuterClass.Status.SUCCESS) {
-                    responseListBuilder.addResponse(Response.responseOK(result.getId()));
-                } else if (result.getStatus() == SinkOuterClass.Status.FALLBACK) {
+            for (SinkOuterClass.SinkResponse result : outputResponses) {
+                if (result.getHandshake().getSot()) {
```

Comment:
what does getSot do?

Reply:
sot means start of the transmission. when set to true, it means the handshake is successful.